Cannot compile Julia 0.6 with libunwind 1.2.1

I am pretty sure this is a bug, but I am not completely sure if its Julia or libunwind the culprit here:

I am trying to compile the 0.6 Julia version but it fails with the following error:

```
from /usr/include/x86_64-pc-linux-gnu/libunwind-x86_64.h:129:0,
                 from /usr/include/x86_64-pc-linux-gnu/libunwind.h:25,
                 from /usr/include/libunwind.h:10,
                 from /var/tmp/portage/dev-lang/julia-0.6.0/work/julia-0.6.0/src/julia_internal.h:642,
                 from /var/tmp/portage/dev-lang/julia-0.6.0/work/julia-0.6.0/src/dlload.c:11:
/var/tmp/portage/dev-lang/julia-0.6.0/work/julia-0.6.0/src/julia_internal.h:646:35: error: floating constant in preprocessor expression
        (UNW_VERSION_MAJOR == 1 && UNW_VERSION_MINOR > 1))
```

It looks like it interpreting the condicional there as a floating number where it should be an integer? If I read it correctly, it looks like a contradiction. Should unwind be at the same time equals to 1 and bigger than 1? 

I am on a GNU/Linux machine (gentoo) with kernel  4.10.4 and gcc 5.4.

--- a/src/julia_internal.h
+++ b/src/julia_internal.h
@@ -625,7 +625,7 @@ extern volatile int jl_in_stackwalk;
 typedef unw_context_t bt_context_t;
 typedef unw_cursor_t bt_cursor_t;
 #  if (!defined(SYSTEM_LIBUNWIND) || UNW_VERSION_MAJOR > 1 ||   \
-       (UNW_VERSION_MAJOR == 1 && UNW_VERSION_MINOR > 1))
+       (UNW_VERSION_MAJOR == 1 && UNW_VERSION_MINOR != 0 && UNW_VERSION_MINOR != 1))
 // Enable our memory manager only for libunwind with our patch or
 // on a newer release
 #    define JL_UNW_HAS_FORMAT_IP 1

